src/internal/connector/exchange/api/mail.go

```diff
@@ -221,6 +225,89 @@ func (c Mail) GetItem(
 	return mail, MailInfo(mail), nil
 }
 
+type mailFolderDeltaPagerer interface {
```

Comment:
We don't need to maintain delta and non-delta versions of these. We only need something that supports enumerating all folders at all levels of the hierarchy efficiently. If the beta API can handle that then it's fine to just switch to that implementation. The delta tokens we get from enumerating mail folders are discarded anyway

Reply:
I kept both since the the new one is a **beta** endpoint and thought it would be better to use the stable version for by default.

Reply:
Removing it.